### PR TITLE
Fix Android build failure due to spaces in `ANDROID_HOME` or `ANDROID_NDK_ROOT` environment variable paths

### LINKED ISSUE #19211: Can't build for Android on Windows 10
Greetings! I'm trying to build Godot for Android using Windows 10, but I get the following problem:

```
MSVC not detected (no VCINSTALLDIR environment variable), attempting MinGW.
Building for Android (armv7) (with NEON)
Could not read source prop file '"C:\Developement\android-ndk-r16b"\source.properties'
Using NDK deprecated headers
Checking for C header file mntent.h... (cached) no
scons: done reading SConscript files.
scons: Building targets ...
[Initial build] Compiling shared ==> platform\android\os_android.cpp
scons: *** [platform\android\os_android.os] O sistema não pode encontrar o arquivo especificado
scons: building terminated because of errors.
```

"O sistema não pode encontrar o arquivo especificado" simply means it cannot find the file. Is there any solution for this? I'm trying to build Godot 3.0.2 and using CMD as admin.

### PR DESCRIPTION
**Pull request by divshekhar**
Issue:
Android builds failed for users with spaces in their paths for `ANDROID_HOME` or `ANDROID_NDK_ROOT` environment variables.

Solution:
Escaped spaces in paths to ensure successful Android builds.

Fixes: #19211 #17710

Please review and let me know if there are any further changes required.

### CODE REVIEW

File: platform/android/detect.py [hunk 1]
```diff
@@ -80,7 +85,7 @@ def install_ndk_if_needed(env: "SConsEnvironment"):
     sdk_root = env["ANDROID_HOME"]
     if not os.path.exists(get_android_ndk_root(env)):
         extension = ".bat" if os.name == "nt" else ""
-        sdkmanager = sdk_root + "/cmdline-tools/latest/bin/sdkmanager" + extension
+        sdkmanager = escape_space_in_path(sdk_root + "/cmdline-tools/latest/bin/sdkmanager" + extension)
```
Comment (akien-mga) [inline platform/android/detect.py:88]: I think it would be more readable to use f-Strings directly, improving the concatenation too.

Comment (divshekhar) [reply platform/android/detect.py:88]: For all the paths or just here?

Comment (m4gr3d) [reply platform/android/detect.py:88]: To be consistent, use it for all paths.

Comment (divshekhar) [reply platform/android/detect.py]: Have made the changes. Kindly let me know if there are any further changes required.

**Review by m4gr3d — APPROVED** [review]
The update looks good to me!

Can you fix the formatting issues.

**Comment by m4gr3d** [comment]
Can you also squash the commits together.

**Pull request closed without merging**